desktop/src/main/kotlin/io/spine/examples/pingh/desktop/Tray.kt

```diff
@@ -91,19 +98,109 @@ internal fun ApplicationScope.Tray(state: AppState) {
     val onClick by rememberUpdatedState(mouseEventHandler(state, menu))
 
     tray = remember {
-        TrayIcon(awtIcon, state.window.title).apply {
+        TrayIcon(icon, state.window.title).apply {
             isImageAutoSize = true
             addMouseListener(onClick)
         }
     }
 
-    SystemTray.getSystemTray().add(tray)
+    SideEffect {
+        if (tray.image != icon) tray.image = icon
+    }
 
     val coroutineScope = rememberCoroutineScope()
-    state.tray
-        .notificationFlow
-        .onEach { tray.displayMessage(it) }
-        .launchIn(coroutineScope)
+
+    DisposableEffect(Unit) {
+        SystemTray.getSystemTray().add(tray)
+
+        state.tray
+            .notificationFlow
+            .onEach { tray.displayMessage(it) }
+            .launchIn(coroutineScope)
+
+        onDispose {
+            SystemTray.getSystemTray().remove(tray)
+        }
+    }
+}
+
+/**
+ * Returns the system tray icon of the application.
+ *
+ * If the user is logged in and there are unread mentions,
+ * a badge showing the number of unread mentions appears on the icon.
+ */
+@Composable
+private fun rememberIcon(state: PinghApplication): Image {
+    val unread by state.unreadMentionCount.collectAsState()
+
+    // We shouldn't use `LocalDensity` here because tray's density doesn't equal it. It
```

Comment:
["It" should be treated better](https://en.wikipedia.org/wiki/Widows_and_orphans).